backend/src/bin/api.rs

```diff
@@ -1,22 +1,42 @@
 use std::error::Error;
 use std::net::{Ipv4Addr, SocketAddr};
+use std::path::PathBuf;
 use std::str::FromStr;
 
+use clap::Parser;
 use sqlx::sqlite::SqliteConnectOptions;
 use sqlx::SqlitePool;
 use tokio::net::TcpListener;
+use tower_http::services::{ServeDir, ServeFile};
 
 use backend::router;
 
 const DB_URL: &str = "sqlite://db.sqlite";
 
+/// Abacus API server
+#[derive(Parser, Debug)]
+struct Args {
+    /// Path to the frontend dist directory to serve through the API server
+    #[arg(short, long)]
+    frontend_dist: Option<PathBuf>,
+}
+
 /// Main entry point for the application, sets up the database and starts the
 /// API server on port 8080.
 #[tokio::main]
 async fn main() -> Result<(), Box<dyn Error>> {
+    let args = Args::parse();
     let pool = create_sqlite_pool().await?;
     let app = router(pool)?;
 
+    let app = if let Some(fd) = args.frontend_dist {
+        app.fallback_service(
+            ServeDir::new(fd.clone()).fallback(ServeFile::new(fd.join("index.html"))),
```

Comment:
Now I'm curious: what's the difference between `not_found_service()` and `fallback()`?

Reply:
`not_found_service()` responds with 404 Not Found, `fallback()` with 200 OK.

Reply:
In which case? If I have an empty `frontend/dist` folder, I get a `404` for both `not_found_service()` and for `fallback()`. If I do have a `frontend/dist/index.html`, I get a `200` for both.

Reply:
With the `not_found_service()`, navigating to `/overview` would serve `index.html` (expected) but with a 404 Not Found status code (not expected). With `fallback()`, this responds with 200 OK instead (expected).